hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ExecutionContext.java

```diff
@@ -83,6 +85,8 @@
     private SnapshotContext snapshotContext;
     private JobConfig jobConfig;
 
+    private final Map<String, Long> jobMetrics = new HashMap<>();
```

Comment:
The maps can be concurrently read while we're updating it. Using `ConcurrentHashMap` isn't feasible either because it can be read just after it was cleared. We need to use volatile field to which we'll assign a built map.

Reply:
Ok, have done this change. Tried to do it as efficiently as possible, but still, it's a constant source of garbage production. Maybe that's not a problem though.